File: cpp/src/IceBox/Exception.cpp
// **********************************************************************
//
// Copyright (c) 2003-present ZeroC, Inc. All rights reserved.
//
// **********************************************************************

#include <IceBox/IceBox.h>

using namespace std;

void
IceBox::FailureException::ice_print(ostream& out) const
{
    Exception::ice_print(out);
    out << ":\nservice failure exception: " << reason;
}

File: js/src/Ice/Exception.js
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

export class Exception extends Error {
    constructor(...params) {
        super(...params);
        this.name = this.constructor._ice_id.substr(2).replace(/::/g, ".");
    }

    ice_id() {
        return this.constructor._ice_id;
    }
}
